akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -184,13 +184,14 @@ object Source {
    * Streams the elements of the given future source once it successfully completes.
    * If the future fails the stream is failed.
    */
-  def fromFutureSource[T, M](future: Future[Graph[SourceShape[T], M]]): javadsl.Source[T, Future[M]] = new Source(scaladsl.Source.fromFutureSource(future))
+  def fromFutureSource[T, M](future: Future[_ <: Graph[SourceShape[T], M]]): javadsl.Source[T, Future[M]] = new Source(scaladsl.Source.fromFutureSource(future))
 
   /**
    * Streams the elements of an asynchronous source once its given `completion` stage completes.
    * If the `completion` fails the stream is failed with that exception.
    */
-  def fromSourceCompletionStage[T, M](completion: CompletionStage[Graph[SourceShape[T], M]]): javadsl.Source[T, CompletionStage[M]] = new Source(scaladsl.Source.fromSourceCompletionStage(completion))
+  def fromSourceCompletionStage[T, M](completion: CompletionStage[_ <: Graph[SourceShape[T], M]]): javadsl.Source[T, CompletionStage[M]] =
+    new Source(scaladsl.Source.fromSourceCompletionStage(completion))
```

Comment:
CompletionStage is invariant and this is the Java API so this definitely needs to be variant at the call site.